Notary v2 Alpha 1 release blog (#48)
notaryproject/notaryproject.dev: content/en/blog/2021/announcing-notation-alpha1.md

---
title: Announcing Notary v2 alpha 1
author:  "[Steve Lasker](https://github.com/stevelasker), Microsoft Azure"
date: 2021-10-20
draft: false
---

We are happy to announce the [alpha 1 release][notation-release] of the [Notary v2 project][notary-v2].
Notary v2 enables signing of all artifacts (Container Images, Software Bill of Materials, Scan Results) stored in an [OCI Distribution][oci-distribution] based registries, with [oras artifacts spec][oras-artifacts] enhancements.
A key tenant of Notary v2 is that it enables promotion of signed artifacts within and across registries, including air-gapped and private network environments.
As artifacts get promoted, users/entities may add new signatures, attesting to the validity of the content for the target environment, enabling a secure supply chain workflow.

Through signing, users choose the artifacts they trust, from the entities they trust, [decoupling location from identity](https://stevelasker.blog/2021/09/24/separating-identity-from-location/).

> ![](./announcing-notation-alpha1/artifact-promotion.svg)

Reviewer: ```suggestion
![](../announcing-notation-alpha1/artifact-promotion.svg)
```

Reviewer: Hugo pushes the page down a level, so you need `..` to access the image.

---

> title: Announcing notation alpha 1

Reviewer: Capitalization
```suggestion
title: Announcing Notation alpha 1
```

> spelling: cSpell:ignore

Reviewer: remove spelling header item
```suggestion
```

> spelling: cSpell:ignore 

Reviewer: we can probably remove the spelling line if you've not got any words in the copy that are triggering the spell checker that shouldn't be.

author:  "[Steve Lasker](https://github.com/stevelasker), Microsoft Azure"
date: 2021-10-20
draft: false
---

> We are happy to announce the [alpha 1 release][notation-release] of the [Notary v2 project][notary-v2].
> Notary v2 enables signing of all artifacts (Container Images, Software Bill of Materials, Scan Results) stored in an [OCI Distribution][oci-distribution] based registries.

Reviewer: Should we refer to "Notation" here, rather than "Notary V2?"
```suggestion
We are happy to announce the [alpha 1 release][notation-release] of the [Notation project][notary-v2].
Notation (formerly named Notary V2) enables signing of all artifacts (Container Images, Software Bill of Materials, Scan Results) stored in [OCI Distribution][oci-distribution] based registries.
```

Author: The project is Notary, the version is 2, and there are specific components with discrete names.
Within the Notary Project, there's
- notation - a cli for sign, push, validate
- notation-go-lib for implementing the sign, push, validate capabilities in other clients
- tuf for enabling notary v2 signing with tuf metadata

> A key tenant of Notary v2 enables promotion of signed artifacts within and across registries, including air-gapped and private network environments.

Reviewer: ```suggestion
A key tenant of Notation is that it enables promotion of signed artifacts within and across registries, including air-gapped and private network environments.
```

> As artifacts are promoted, users/entities may add new signatures, attesting to the validity of the content for the target environment, enabling a secure supply chain workflow.

Reviewer: ```suggestion
As artifacts get promoted, users/entities may add new signatures, attesting to the validity of the content for the target environment, enabling a secure supply chain workflow.
```

Through signing, users choose the artifacts they trust, from the entities they trust, [decoupling location from identity](https://stevelasker.blog/2021/09/24/separating-identity-from-location/).

![](../announcing-notation-alpha1/artifact-promotion.svg)

## Features

> The [notation alpha 1 release][notation-release] supports the following [Notary v2 goals][notary-v2-goals]:

Reviewer: ```suggestion
The [Notation alpha 1 release][notation-release] supports the following [Notation goals][notary-v2-goals]:
```

- Offline signature creation
- Signatures attesting to authenticity and/or certification
- Maintain the original artifact digest and collection of associated tags, supporting existing dev through deployment workflows
- Multiple signatures per artifact, enabling the originating vendor signature, public registry certification and user/environment signatures
- Signature persistance within an [OCI Artifact][oci-artifacts] enabled, [distribution-spec][oci-distribution] based registry, with the [oras artifacts spec][oras-artifacts] enhancements.
- Air-gapped environments, where the originating registry of content is not accessible

> - Artifact and signature copying within and across [OCI Artifact][oci-artifacts] enabled, [distribution-spec][distribution-spec] based registries, using [oras copy][oras-copy]

Reviewer 2: oras-copy reference needs to be removed.

- Verification of signatures, through a configuration based policy

> Future versions of Notary v2 will support:

Reviewer: ```suggestion
Future versions of Notation will support:
```

---
title: Announcing Notary v2 alpha 1
author:  "[Steve Lasker](https://github.com/stevelasker), Microsoft Azure"
date: 2021-10-20
draft: false
---

We are happy to announce the [alpha 1 release][notation-release] of the [Notary v2 project][notary-v2].

> Notary v2 enables signing of all artifacts (Container Images, Software Bill of Materials, Scan Results) stored in an [OCI Distribution][oci-distribution] based registries, with [oras artifacts spec][oras-artifacts] enhancements.

Reviewer: ```suggestion
Notary v2 enables signing of all artifacts (Container Images, Software Bill of Materials, Scan Results) stored in [OCI Distribution][oci-distribution] based registries, with [oras artifacts spec][oras-artifacts] enhancements.
```

or

```suggestion
Notary v2 enables signing of all artifacts (Container Images, Software Bill of Materials, Scan Results) stored in an [OCI Distribution][oci-distribution] based registry, with [oras artifacts spec][oras-artifacts] enhancements.
```

A key tenant of Notary v2 is that it enables promotion of signed artifacts within and across registries, including air-gapped and private network environments.

In addition to signature promotion, Notary v2 focuses on ease of use, with minimal dependencies. While Notary v2 can integrate with other supply chain efforts, there are no additional services required to sign or validate an artifact.
As artifacts get promoted, users/entities may add new signatures, attesting to the validity of the content for the target environment, enabling a secure supply chain workflow.

Through signing, users choose the artifacts they trust, from the entities they trust, [decoupling location from identity](https://stevelasker.blog/2021/09/24/separating-identity-from-location/).

> ![](../announcing-notation-alpha1/artifact-promotion.svg)

Reviewer: The image has some text that is cropped in various boxes.

There is also an unreadable icon in the upper right corner.

<img width="1869" alt="Screen Shot 2021-10-21 at 1 25 53 PM" src="https://user-images.githubusercontent.com/4453979/138351832-a103f162-1565-4539-93d6-b38d36f3fad9.png">

These aren't showstopper issues, but we may want to open an issue to fix the image after the PR is merged.

## Features

The [Notation alpha 1 release][notation-release] supports the following [Notary v2 goals][notary-v2-goals]:
- Offline signature creation
- Signatures attesting to authenticity and/or certification
- Maintain the original artifact digest and collection of associated tags, supporting existing dev through deployment workflows
- Multiple signatures per artifact, enabling the originating vendor signature, public registry certification and user/environment signatures
- Signature persistance within an [OCI Artifact][oci-artifacts] enabled, [distribution-spec][oci-distribution] based registry, with [oras artifacts spec][oras-artifacts] enhancements
- Air-gapped environments, where the originating registry of content is not accessible

> - Artifact and signature copying within and across [OCI Artifact][oci-artifacts] enabled, [distribution-spec][distribution-spec] based registries

Reviewer: `[distribution-spec]` is not listed in the links section below, so this link is not rendering correctly.

We'll either need to add the link below, or remove the hyperlink here.

- Verification of signatures, through a configuration based policy

> Future versions of Notary v2 will support:

Reviewer: Since the previous paragraph lists the features as being a part of the "Notation alpha 1 release, supporting Notary V2 goals," I think that this line should read:

```suggestion
Future versions of Notation will support:
```

The goals of Notary V2 aren't changing, just the Notation implementation. If my understanding here is incorrect, then maybe the line should read:

```suggestion
Future goals of Notary v2 will include:
```

- Key hierarchies and delegation
- Key revocation, including private and air-gapped registries
- Verification through policy, enabling environment specific validations
- OCI Distribution 1.0 support

## Getting Started

> 

Reviewer: I'd suggest adding a line here to introduce this section. Something simple like "Here are some resources to help get started with Notation and Notary V2" (just so we don't have a header leading directly into a bulleted list).
